sw/device/silicon_creator/lib/drivers/mock_otp.h

```diff
@@ -0,0 +1,46 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_MOCK_OTP_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_MOCK_OTP_H_
+
+#include "sw/device/lib/testing/mask_rom_test.h"
+#include "sw/device/silicon_creator/lib/drivers/otp.h"
+
+namespace mask_rom_test {
+namespace internal {
+
+/**
+ * Mock class for otp.c.
+ */
+class MockOtp {
+ public:
+  MOCK_METHOD(uint32_t, read32, (uint32_t address));
+  MOCK_METHOD(uint32_t, read64, (uint32_t address));
+  MOCK_METHOD(void, read, (uint32_t address, uint32_t *data, size_t num_words));
+  virtual ~MockOtp() {}
+};
```

Comment:
There is also a MockOtp implementation in `shutdown_unittest.cc`.  I think it should be migrated to use this implemenation.

However, the version in shutdown_unittest uses `NiceMock` and implements a bit more generic of an interface over the SW_CFG partition (ie: it has a struct declared representing the entire SW_OWNER portion of OTP).

We can unify the otp mocking in a follow-up PR if you prefer.

Reply:
I'll handle this in a separate PR as we discussed.